extensions/interactions/MathEquationInput/directives/math-equation-input-rules.service.spec.ts

```diff
@@ -134,6 +134,107 @@ describe('Math equation input rules service', () => {
+      'y^2=m*x*y+c*y', {x: inputString, y: positionOfTerms})).toBeFalse();
+
+    positionOfTerms = 'rhs';
+    // Accepted cases.
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y=m*x+c', {x: inputString, y: positionOfTerms})).toBeTrue();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y^2=m*x+c', {x: inputString, y: positionOfTerms})).toBeTrue();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      '0=x*m+c', {x: inputString, y: positionOfTerms})).toBeTrue();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      '0=c+m*x', {x: inputString, y: positionOfTerms})).toBeTrue();
+    // Rejected cases.
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y-m*x=c', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'm*x+c=y', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y^2=m*x', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'x=m*y+c', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y/(m*x+c)=1', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y^2=m*x*y+c*y', {x: inputString, y: positionOfTerms})).toBeFalse();
+
+    positionOfTerms = 'both';
+    // Accepted cases.
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y=m*x+c',
+      {x: inputString, y: positionOfTerms})).toBeTrue();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y=c+m*x', {x: inputString, y: positionOfTerms})).toBeTrue();
+    // Rejected cases.
+    expect(meirs.MatchesUpToTrivialManipulations(
+      '(y^2)/y=m*x+c', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y-m*x=c', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'm*x+c=y', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y=m*x', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'x=m*y+c', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y/(m*x+c)=1', {x: inputString, y: positionOfTerms})).toBeFalse();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y^2=m*x*y+c*y', {x: inputString, y: positionOfTerms})).toBeFalse();
+
+    positionOfTerms = 'irrelevant';
+    // Accepted cases.
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y=m*x+c', {x: inputString, y: positionOfTerms})).toBeTrue();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y=c+m*x', {x: inputString, y: positionOfTerms})).toBeTrue();
+    expect(meirs.MatchesUpToTrivialManipulations(
+      'y-m*x-2*c=-c', {x: inputString, y: positionOfTerms})).toBeTrue();
```

Comment:
Hmm, why is this one true? /cc @BenHenning for confirmation.

Reply:
This shouldn't match. Trivial manipulations doesn't include rearranging around the equals sign (both sides must match their respective sides according to the rules for algebraic expressions, those are, associative and commutative reordering).

Reply:
Ah, yes, this is happening since we need to manipulate the sides (multiply by -1 and add to the other side), and using nerdamer causes unwanted simplifications.
One way that comes to mind would be to split each side into terms and then for every term in the answer, check if it exists on the same side, or if its negation exists on the other side. But, in this case, as well we would have to multiply the term by -1 and nerdamer would simplify values within that term. I don't see a way of matching this without performing operations.
Any suggestions?

Reply:
Yup, that works, done, thanks!